wkcuber/api/View.py

```diff
@@ -156,6 +161,40 @@ def _check_chunk_size(self, chunk_size):
                 f"The passed parameter 'chunk_size' {chunk_size} must be a multiple of (32, 32, 32)."
             )
 
+    def _handle_compressed_write(self, absolute_offset, data):
+        if (
+            self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4
+            or self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4HC
+        ):
+            # data is compressed
+
+            # calculate aligned bounding box
+            file_bb = np.full(3, self.header.file_len * self.header.block_len)
+            margin_to_top_left = np.array(absolute_offset) % file_bb
+            aligned_offset = np.array(absolute_offset) - margin_to_top_left
+            bottom_right = np.array(absolute_offset) + np.array(list(data.shape)[-3:])
+            margin_to_bottom_right = file_bb - (bottom_right % file_bb)
+            aligned_bottom_right = bottom_right + margin_to_bottom_right
+            aligned_shape = aligned_bottom_right - aligned_offset
+
+            if tuple(aligned_offset) != tuple(absolute_offset) or tuple(
+                aligned_shape
+            ) != tuple(list(data.shape)[-3:]):
+                # the data is not aligned
+                # read the aligned bounding box
+                aligned_data = self.read(offset=aligned_offset, size=aligned_shape)
```

Comment:
I was also wondering about this, but I think it's fine to leave this to the user.

Reply:
Do you have an idea on how to solve this with a warning elegantly and without a lot of overhead? Maybe a small cache about the last write could already be enough.

Reply:
What do you think about adding another parameter to the write function? For example, `write(..., allow_compressed_write=True)`. By default this parameter would be `False`  (and if the target is compressed throw an exception which tells the user to either align the input data or use that parameter with the note of the performance hit).
I think, making this opt-in would be reasonable, since it should only be done when one is sure that the performance hit is not critical. Emitting a warning won't work, in my opinion, since it won't be read mostly.

Reply:
Sounds great!
![](https://media.giphy.com/media/UtKNpG63sD01cpGppg/giphy.gif)